tests/ui/test_other_user.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import os
+import pytest
+
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.ui.mcg_ui import ObcUI
+from ocs_ci.ocs import ocp, constants
+from ocs_ci.helpers.helpers import create_unique_resource_name
+from ocs_ci.ocs.ui.views import locators
+from ocs_ci.framework.testlib import ManageTest, ui, bugzilla, polarion_id, tier2
+from time import sleep
+from ocs_ci.utility.utils import ceph_health_check
+
+logger = logging.getLogger(__name__)
+
+
+class ObcUi(ObcUI):
+    def __init__(self, driver):
+        super().__init__(driver)
+        self.sc_loc = locators[self.ocp_version]["obc"]
+
+    def check_obc_option(self, text="Object Bucket Claims"):
+        """check OBC is visible to user after giving admin access"""
+
+        sc_name = create_unique_resource_name("namespace-", "interface")
+        self.do_click(self.sc_loc["Developer_dropdown"])
+        self.do_click(self.sc_loc["select_administrator"], timeout=5)
+        self.do_click(self.sc_loc["create_project"])
+        self.do_send_keys(self.sc_loc["project_name"], sc_name)
+        self.do_click(self.sc_loc["save_project"])
+        self.choose_expanded_mode(mode=True, locator=self.page_nav["Storage"])
+        obc_found = self.wait_until_expected_text_is_found(
+            locator=self.sc_loc["obc_menu_name"], expected_text=text, timeout=10
+        )
+        if not obc_found:
+            logger.info("user is not able to access OBC")
+            self.take_screenshot()
+            return None
+        else:
+            logger.info("user is able to access OBC")
+
+        namespace_obj = OCP(kind=constants.NAMESPACE, resource_name=sc_name)
+        namespace_obj.delete(resource_name=sc_name)
+
+
+class TestOBCUi(ManageTest):
+    """
+    Validate User able to see the OBC resource from the Console
+
+    Test Process:
+    1.Created a user
+    2.Create project
+    3.Added admin role to this user of the project.
+    4.Validated the access of OBC from Console.
+
+    """
+
+    @pytest.fixture(scope="function", autouse=True)
+    def teardown(self, request):
+        def finalizer():
+            logger.info("Perform Ceph health checks ")
+            ceph_health_check()
+
+        request.addfinalizer(finalizer)
+
+    @ui
+    @tier2
+    @bugzilla("2031705")
+    @polarion_id("OCS-4620")
+    def test_create_storageclass_rbd(self, user_factory, login_factory):
+        """Create user"""
+        user = user_factory()
+
+        kubeconfig = os.getenv("KUBECONFIG")
+        kube_data = ""
+        with open(kubeconfig, "r") as kube_file:
+            kube_data = kube_file.readlines()
+        sleep(30)
+        with open(kubeconfig, "w+") as kube_file:
```

Comment:
Can you explain why you are reading the kubeconfig and re-writing it?
This could affect all other tests in the run so we don't re-write kubeconfig in mid run.

Reply:
actually i was trying diffrent way to login user so i used kubeconfig. so Not needed now .Will remove this code.